cmd/platform/server.go

```diff
@@ -78,6 +78,8 @@ func runServer(configFileLocation string, disableConfigWatch bool) {
 		a.LoadLicense()
 	}
 
+	a.DoAdvancedPermissionsMigration()
```

Comment:
Will we get a chance to remove this in the future, or does it need to be here forever in case somebody hasn't yet performed the upgrade?

Reply:
I think this is here forever sadly.

Reply:
It would be nice if we could have this called from the DB upgrade code. That way if we ever remove the ability for a future version of MM to do a full upgrade we can remove this. (We have done that before. You cant upgrade from any less that 3.6 currently)

Reply:
I did try to do that originally, but it got *very* ugly and still didn't even work. The reason is that the migration depends on the license, which isn't initialised at the point at which the database migrations run, as it lives in the database.

Reply:
Can we put a comment in the db upgrade code for the specific release.  A lot of users/customers look at the upgrade script and pre-upgrade their DB.

I wonder if we could give a script or tool to do this ahead of time? (0/5)

Reply:
Opened XYZ-78 to add the note when we get closer to master merge.